src/Sarif/Writers/ConsoleLogger.cs

```diff
@@ -322,6 +324,12 @@ static internal string FormatNotificationMessage(Notification notification, stri
 
             sb.Append(notification.Message.Text);
 
+            if (notification.Exception != null)
+            {
+                sb.AppendLine();
+                sb.AppendLine(JsonConvert.SerializeObject(notification.Exception));
```

Comment:
>JsonConvert.SerializeObject(notification.Exception) [](start = 30, length = 51)

Since Exception isn't an exception class, its an ExceptionData class from sarif, it does not expose a ToString which generate all the fields. With that in mind, I switched to serializing it.

Reply:
Let's write an ExceptionData.ToString() implementation here that mirrors System.Exception.ToString() behavior

---
In reply to: [551626348](https://github.com/microsoft/sarif-sdk/pull/2217#discussion_r551626348) [](ancestors = 551626348)

Reply:
done! thanks for guiding me!

---
In reply to: [551649906](https://github.com/microsoft/sarif-sdk/pull/2217#discussion_r551649906) [](ancestors = 551649906,551626348)